### ISSUE TITLE
Adding store

### ISSUE THREAD

**Issue by Narrowcastingscherm** [thilojaeggi/WinGetty#12]
So i have build the wingetty enviroment and all is https. When i try to add the store to winget inget an error failed to open the added source. Winget api is running.

**Comment by thilojaeggi** [comment]
Just to check, if you open the WinGet webinterface in a browser the https certificate is regarded as valid, not self-signed?
Registration can be disabled in the docker_compose file using the Env variable

**Comment by bjoernroos** [comment]
I have the same problem with http:
Error opening added source.

**Comment by thilojaeggi** [comment]
> I have the same problem with http:
> 
> Error opening added source.

You need to use https

**Comment by bjoernroos** [comment]
How can i add/activate https?

**Comment by thilojaeggi** [comment]
If you open the dashboard in a browser do you see a lock/valid certificate?

**Comment by bjoernroos** [comment]
Thx. It is sometimes better to read the entire documentation.
;)

I have add the source to winget and i can search for packets.
Now i want to use to WingetUI. But when i add the source i get an 404 error

**Comment by thilojaeggi** [comment]
WingetUI might only support the normal source type and not Microsoft.Rest

**Comment by bjoernroos** [comment]
In the Post:
https://github.com/marticliment/WingetUI/issues/734
The Rest Api should be supported.

**Comment by excelsi** [comment]
Only short question regarding the Certificate Setup.
I don't know how the setup Nginx etc as HTTPS Reverse Proxy etc.
Is there any method to add an Web Server Certificate directly in the Container to the Web Server and how can do that?
Thanks!

**Comment by thilojaeggi** [comment]
You could (for quick testing) clone the repo and run basically all the setup commands from the Dockerfile then modify start.sh to use your own (trusted by clients) certificate.

Something like this `gunicorn --certfile=server.crt --keyfile=server.key --bind 0.0.0.0:443 "app:create_app"`
should technically work.

**Comment by excelsi** [comment]
@thilojaeggi 
I have now managed the HTTPS/HTTP Redirect with Windows Web Application Proxy:
![image](https://github.com/thilojaeggi/WinGetty/assets/8747559/245a9dc5-e9d1-43c2-8577-e1bdc594e26c)
I added the source sucessfully to winget and i could also search the apps i published.
But if i want to install a msix package the store gives me back the Backend Server URL with HTTPS for download (which is not accessible) 
![image](https://github.com/thilojaeggi/WinGetty/assets/8747559/739dc372-88ce-4151-9ad1-9688cb7973c8)
I tried two Options on the Web Application Proxy regarding HTTP Redirect but it didn't fix the problem.
Is there any possibility to reconfigure this inside the App?
Thanks!

**Comment by thilojaeggi** [comment]
I'm pretty sure this is the same issue as here #17 with your proxy not forwarding the host variable correctly.

**Comment by Narrowcastingscherm** [comment]
What is going wrong, i follewed all the instructions but still i get an error

![Schermafbeelding 2023-10-09 131520](https://github.com/thilojaeggi/WinGetty/assets/67043140/c56bcc0d-6597-4642-b46a-e803843924e0)

**Comment by thilojaeggi** [comment]
Just to check, the client trusts the certificate and the url under the Setup tab is the same as you've entered?

**Comment by Narrowcastingscherm** [comment]
yes cert is trusted and the link i copied from the setup page directly, i only made it black because of privacy inside company

**Comment by Narrowcastingscherm** [comment]
is there something i can try?

**Comment by Narrowcastingscherm** [comment]
i use proxymanager for the reverse proxy on a local domain, winget server is not going outside and certs are generated on local server

**Comment by rileyz** [comment]
@Narrowcastingscherm I have WinGetty working with WinGetUI (running on Ubuntu server with stunnel for https/ssl).  I had issues like you, when trying to add my private repo via the WinGetUI GUI (ironically).  I was never able to via the repo via GUI method.  But if you add the source via command line, then it will be detected by WinGetUI. 

I was on a Window based system, so adjust as per your needs.
Method: Find WinGetUI's private winget.exe, once found, use the command below to add the repo.
`winget.exe source add -n WinGetty -t "Microsoft.Rest" -a https://YourURLToWinGetty.com/wg/`

In regards to https/ssl, check out stunnel, simple and quick solution for https if you just want to get up and running to test WinGetty.

**Comment by rileyz** [comment]
@Narrowcastingscherm Very unhelpful comment here, but I just wanted to confirm to you the combination does work.  WinGetty working with WinGetUI.

I was unable to add the repo via the WinGetUI GUI, kept erroring. I was able to add it via the command line, and it did show up in WinGetUI. I was able to install packages etc.

My setup was much simpler. Docker running on Ubuntu server. I used stunnel for the https/ssl. Reused a duckdns cert from another server (privkey and fullchain) for stunnel.

**Comment by excelsi** [comment]
@thilojaeggi would this be one valid config in the docker compose file?
<img width="468" alt="image" src="https://github.com/thilojaeggi/WinGetty/assets/8747559/05a110e5-9679-49d4-a5f4-35e35696a9fd">

Thanks!

**Comment by bjoernroos** [comment]
> @Narrowcastingscherm Very unhelpful comment here, but I just wanted to confirm to you the combination does work. WinGetty working with WinGetUI.
> 
> I was unable to add the repo via the WinGetUI GUI, kept erroring (WinGetUI issue). I was able to add it via the command line, and it did show up in WinGetUI. I was able to install packages etc.
> 
> My setup was much simpler. Docker running on Ubuntu server. I used stunnel for the https/ssl. Reused a duckdns cert from another server (privkey and fullchain) for stunnel.

I have add the wingetty repo to winget. But it will not show in wingetui. Is there a command-line command add it to wingetui?

**Comment by rileyz** [comment]
> I have add the wingetty repo to winget. But it will not show in wingetui. Is there a command-line command add it to wingetui?

Keep in mind I'm a WinGet newbie and just wanted to play around with WinGetty for private repo reasons. Literally only had 5 days use/play. 

On the machine with WinGetUI, I used the (private?) winget.exe that came bundled with WinGetUI, no other winget was installed.  I used the winget in path `"C:\Program Files\WingetUI\PackageManagers\winget-cli_x64"`.  When the repo was added to WinGetUI, I found the data pertaining to the repo was saved in `"C:\Users\<username>\AppData\Local\Microsoft\WinGet\Settings\defaultState\user_sources"`, which a presume is the same for all wingets(?) 🤷‍♂️  From there it showed up in WinGetUI and I could install my test MSIX package.

![wingetui](https://github.com/thilojaeggi/WinGetty/assets/8976376/2eeeaa31-4403-477e-ac23-b1f47cdfbe9c)
